blackbox/docs/general/ddl/data-types.rst

```diff
@@ -197,75 +197,95 @@ Example::
     ... values ('localhost', 'not.a.real.ip');
     SQLActionException[ColumnValidationException: Validation failed for ip_addr: Cannot cast 'not.a.real.ip' to type ip]
 
-.. _data-type-timestamp:
+.. _date-time-types:
 
-``timestamp``
-=============
-
-The timestamp type is a special type which maps to a formatted string.
-Internally it maps to the UTC milliseconds since ``1970-01-01T00:00:00Z`` stored
-as ``bigint``. Timestamps are always returned as ``bigint`` values.
+Date/Time Types
+===============
 
-The default format is dateOptionalTime_ and cannot be changed currently.
++------------------------------+---------+------------------------------+------------------------------------+
+| Name                         | Size    | Description                  | Range                              |
++==============================+=========+==============================+====================================+
+| ``timestamp with time zone`` | 8 bytes | time and date with time zone | ``292275054BC`` to ``292278993AD`` |
```

Comment:
This new range contradicts what we had documented earlier. Which is true?

Reply:
This range is taken from what we had documented previously, see [timestamp](https://github.com/crate/crate/blob/master/blackbox/docs/general/ddl/data-types.rst#timestamp)
```
Due to internal date parsing, not the full ``bigint`` range is supported for
timestamp values, but only dates between year ``292275054BC`` and
``292278993AD``, which is slightly smaller.
```

Reply:
Ah right, I somehow implied that by starting with 1970 for the internal storage it can't go below - but it will simply be negative.